File: results/py/COUNT_NUMBERS_THAT_DONT_CONTAIN_3.cpp
int f_filled ( int n ) {
  if ( n < 3 ) return n ;
  else if ( n >= 3 && n < 10 ) return n - 1 ;
  int po = 1 ;
  while ( n / po > 9 ) po = po * 10 ;
  int msd = n / po ;
  if ( msd != 3 ) return f_filled ( msd ) * f_filled ( po - 1 ) + f_filled ( msd ) + f_filled ( n % po ) ;
  else return f_filled ( msd * po - 1 ) ;
}

File: TransCoder_tokenized_test_set_functions/python/COUNT_NUMBERS_THAT_DONT_CONTAIN_3.py
def count ( n ) :
    if n < 3 :
        return n
    elif n >= 3 and n < 10 :
        return n - 1
    po = 1
    while n / po > 9 :
        po = po * 10
    msd = n / po
    if msd != 3 :
        return count ( msd ) * count ( po - 1 ) + count ( msd ) + count ( n % po )
    else :
        return count ( msd * po - 1 )
    
